WickedEngine/wiECS.h

```diff
@@ -161,6 +191,34 @@ namespace wi::ecs
 			}
 		}
 
+		//Read/write one single component onto an archive, make sure entity are serialized first
+		inline void Component_Serialize(Entity entity, wi::Archive& archive, EntitySerializer& seri)
```

Comment:
Instead of this, it would be better to have a function on the Library, called Entity_Serialize(). In that case, the `archive.IsReadMode()` doesn't need to be checked for every component, but instead:
```cpp
if(archive.IsReadMode())
{
  for(all component mangers)
  {
    read...
  }
}
else
{
  for(all component mangers)
  {
    write...
  }
}
```

Reply:
For this, the thing is that I can't get the specific type of the component since I've made the componentManagers into an interface.

Reply:
Can't directly access the component data so what I did here is to split Component_Serialize into two, and then serializing it like above:
```
if(archive.IsReadMode())
{
  for(all component mangers)
  {
    Component_Serialize_R...
  }
}
else
{
  for(all component mangers)
  {
    Component_Serialize_W...
  }
}
```

Reply:
Oh, I didn't see that it can't be done. In this case I like the previous version better, sorry.

Reply:
@turanszkij also how about the archive version inclusion within the ComponentLibrary?

Reply:
Looks good, but I'll want to test it

Reply:
Ok, done.